### DISCUSSION TITLE
viper.ReadInConfig is not using the correct file type set with viper.SetConfigType

### DISCUSSION THREAD

**Discussion by ssi-anik** [spf13/viper#1716] [Q&A]
Viper version: **v1.18.1**
Golang Version: **go1.21.4 darwin/arm64**

Okay, I have read https://github.com/spf13/viper/issues/1408 and it solves the problem.

But going through the code I can see

Viper prepares the filename and checks the file type to see if the type is supported or not.
https://github.com/spf13/viper/blob/0b0a1104ba9ab05dd66ca57b15ac3836131d30c1/viper.go#L1623-L1630

When preparing the filename, it checks in the order the `SupportedExts` is defined.
https://github.com/spf13/viper/blob/0b0a1104ba9ab05dd66ca57b15ac3836131d30c1/file.go#L27-L44

So, if I have a **.json** file and **.yaml** file in the same directory, it will read the **json** as it comes first in the supported extensions list.

Even though it checks whether the extension is supported or not, it just ignores that file with the specified file type.
So, it will use **configuration.json** over **configuration.yml** for the following scenario.

```golang
v.SetConfigName(fileName) // "configuration"
v.SetConfigType(fileType) // "yaml"
v.AddConfigPath(filePath) // "."
```

Files:
- configuration.yaml
- configuration.json

Maybe this check is done for `SetConfigFile`.
https://github.com/spf13/viper/blob/0b0a1104ba9ab05dd66ca57b15ac3836131d30c1/viper.go#L1628-L1630

Maybe you can suggest `viper.SupportedExts = []string{"yaml", "yml"}` or maybe I am wrong!

But can't it be done to use the specified type without this tweak?

TIA.

**Comment by AuZhoomLee** [comment]
#1840
